contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+            address(this), // owner
+            msg.sender, // address to transfer to
+            order.collateral, // withdrawn asset
+            order.vaultID, // vaultId
+            _amount, // amount
+            0, //index
+            "" //data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit CollateralWithdrawn(_orderID, _amount);
+    }
+
+    /**
+     * @notice Deposits the `asset` from msg.sender without an approve
+     * `v`, `r` and `s` must be a valid `secp256k1` signature from `owner`
+     * over the EIP712-formatted function arguments
+     * @param _acct signer account
+     * @param _asset is the asset address to deposit
+     * @param _amount is the amount to deposit
+     * @param _deadline must be a timestamp in the future
+     * @param _v is a valid signature
+     * @param _r is a valid signature
+     * @param _s is a valid signature
+     */
+    function _depositWithPermit(
+        address _acct,
+        address _asset,
+        uint256 _amount,
+        uint256 _deadline,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) private {
+        require(_amount > 0, "OTCWrapper: amount cannot be 0");
+
+        // Sign for transfer approval
+        IERC20Permit(_asset).permit(_acct, address(this), _amount, _deadline, _v, _r, _s);
+
+        // An approve() by the msg.sender is required beforehand
+        IERC20(_asset).safeTransferFrom(_acct, address(this), _amount);
+    }
+
+    /************************************************
+     *  OTC OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice places an order
+     * @param _underlying underlying asset address
+     * @param _isPut option type the vault is selling
+     * @param _strikePrice option strike price
+     * @param _expiry option expiry timestamp
+     * @param _premium order premium amount
+     * @param _notional order notional
+     */
+    function placeOrder(
+        address _underlying,
+        bool _isPut,
+        uint256 _strikePrice,
+        uint256 _expiry,
+        uint256 _premium,
+        uint256 _notional
+    ) external {
+        require(
+            _notional > minMaxNotional[_underlying].min && _notional < minMaxNotional[_underlying].max,
+            "OTCWrapper: invalid notional value"
+        );
+        require(_expiry > block.timestamp, "OTCWrapper: expiry must be in the future");
+
+        latestOrder += 1;
+
+        orders[latestOrder] = Order(
+            _underlying,
+            address(0),
+            _isPut,
+            _strikePrice,
+            _expiry,
+            _premium,
+            _notional,
+            _msgSender(),
+            address(0),
+            0,
+            address(0),
+            block.timestamp
+        );
+
+        emit OrderPlaced(latestOrder, _underlying, _isPut, _strikePrice, _expiry, _premium, _notional, _msgSender());
```

Comment:
latestOrder - 1

Reply:
discontinued - latest order and event number can be equal: eg. 
- latestOrder= 1 | event number = 1  Vs latestOrder= 1 | event number = 0